File: server/apps/nexotype/schemas/user/pathway_score_schemas.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime

# ==========================================
# Request Schemas
# ==========================================

class PathwayScoreCreate(BaseModel):
    """Schema for creating a new pathway score"""
    subject_id: int
    pathway_id: int
    score: float
    calculated_at: datetime


class PathwayScoreUpdate(BaseModel):
    """Schema for updating a pathway score"""
    subject_id: int = None
    pathway_id: int = None
    score: float = None
    calculated_at: datetime = None


# ==========================================
# Response Schemas
# ==========================================

class PathwayScoreDetail(BaseModel):
    """Schema for pathway score details"""
    id: int
    subject_id: int
    pathway_id: int
    score: float
    calculated_at: datetime
    created_at: datetime
    updated_at: datetime | None = None

    model_config = ConfigDict(from_attributes=True)


class PathwayScoreListResponse(BaseModel):
    """Response schema for listing pathway scores"""
    success: bool
    data: list[PathwayScoreDetail] | None = None
    count: int = 0
    error: str | None = None


class PathwayScoreResponse(BaseModel):
    """Response schema for single pathway score operations"""
    success: bool
    data: PathwayScoreDetail | None = None
    error: str | None = None


class MessageResponse(BaseModel):
    """Simple message response"""
    success: bool
    message: str | None = None
    error: str | None = None

File: client/src/modules/nexotype/schemas/user/pathway-score.schemas.ts
import { z } from 'zod';

// ==========================================
// Full Schema (GET responses)
// ==========================================

export const PathwayScoreSchema = z.object({
  id: z.number(),
  subject_id: z.number().int(),
  pathway_id: z.number().int(),
  score: z.number(),
  calculated_at: z.string(),
  created_at: z.string(),
  updated_at: z.string().nullable(),
});

// ==========================================
// Input Schemas (POST/PUT)
// ==========================================

export const CreatePathwayScoreSchema = z.object({
  subject_id: z.number().int(),
  pathway_id: z.number().int(),
  score: z.number(),
  calculated_at: z.string().min(1),
});

export const UpdatePathwayScoreSchema = z.object({
  subject_id: z.number().int().optional(),
  pathway_id: z.number().int().optional(),
  score: z.number().optional(),
  calculated_at: z.string().optional(),
});

// ==========================================
// Types
// ==========================================

export type PathwayScore = z.infer<typeof PathwayScoreSchema>;
export type CreatePathwayScore = z.infer<typeof CreatePathwayScoreSchema>;
export type UpdatePathwayScore = z.infer<typeof UpdatePathwayScoreSchema>;

// ==========================================
// Response Types
// ==========================================

export type PathwayScoreResponse = {
  success: boolean;
  data?: PathwayScore;
  error?: string;
};

export type PathwayScoresResponse = {
  success: boolean;
  data?: PathwayScore[];
  count?: number;
  error?: string;
};
